docs/src/usage.md

```diff
@@ -184,6 +184,20 @@ Dict{String,Any} with 1 entry:
   "data" => [0.166881, 0.817174, 0.413097, 0.955415, 0.139473, 0.49518, 0.416731, 0.431096, 0.126912, 0.600469]
 ```
 
+#### Tag Context
+
+It is possible to introduce a tag context, such that all the [`checkpoint`](@ref) operations inside the context are tagged with the given tags.
+The context is [dynamically scoped](https://en.wikipedia.org/wiki/Scope_(computer_science)#Lexical_scope_vs._dynamic_scope_2), meaning that the tags persist through function calls.
+Tag contexts can be nested and can be used inside the package as well as in the application.
+
+For example:
+```julia
+with_tags(:tag=>1, :othertag=>2) do
```

Comment:
Can we have a different name for this? Maybe just `Checkpoints.tags`?

Reply:
I also like the `with`, Julia has `withenv` and `with_logger`.

Might want `with_checkpoint_tags`, since a tags doesn't give much context (pun not intended).
Or we could not export it and make the user doe `Checkpoints.with_tags` but I would want a good reason to not export the public API